### PR TITLE
copy/paste SOCKS for Firefox

### PR DESCRIPTION
**Pull request by trevj**
Related issue:
https://github.com/uProxy/uproxy/issues/419

This makes copy/paste SOCKS run on Firefox. A bunch of refactoring was required -- to keep changes minimal, I extracted out into `glue.js` a function that loads the freedomjs module. Firefox also needs a little love when it comes to message passing. What I've done is similar to the frontend, which enumerates all possible commands then sets up `on` and `emit` forwarders.

[Review on Reviewable](https://reviewable.io/reviews/uproxy/uproxy-lib/201)

### CODE REVIEW

File: src/samples/copypaste-socks-firefoxapp/lib/main.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    'gotPeerSDP',
+    'gatherMessage',
+    'giveWithSDP',
+    'publicKeyExport',
+    'ciphertext',
+    'verifyDecryptResult',
+    'bytesReceived',
+    'bytesSent',
+    'proxyingStarted',
+    'proxyingStopped'];
```
Comment (bemasc) [inline src/samples/copypaste-socks-firefoxapp/lib/main.js]: Nit: `];` on the next line here and below.

Comment (trevj) [inline src/samples/copypaste-socks-firefoxapp/lib/main.js]: Done!

**Comment by trevj** [comment]
A few things I forgot to mention:
 * 
 * tested manually and via the `connect-pair.py`

**Comment by trevj** [comment]
I asked @lucyhe to sanity check the Firefox stuff. Thanks @lucyhe !

**Comment by lucyhe** [comment]
looks good !! :+1:

**Pull request merged**